src/test/java/staffconnect/logic/commands/MarkCommandTest.java

```diff
@@ -0,0 +1,120 @@
+import static staffconnect.logic.commands.CommandTestUtil.assertCommandFailure;
+import static staffconnect.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static staffconnect.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static staffconnect.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static staffconnect.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static staffconnect.testutil.TypicalPersons.getTypicalStaffBook;
+
+import org.junit.jupiter.api.Test;
+
+import staffconnect.commons.core.index.Index;
+import staffconnect.logic.Messages;
+import staffconnect.model.Model;
+import staffconnect.model.ModelManager;
+import staffconnect.model.StaffBook;
+import staffconnect.model.UserPrefs;
+import staffconnect.model.person.Person;
+import staffconnect.testutil.PersonBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for {@code MarkCommand}.
+ */
+public class MarkCommandTest {
+
+    private Model model = new ModelManager(getTypicalStaffBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Person personToMark = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person markedPerson = new PersonBuilder(personToMark).withFavourite(true).build();
+        MarkCommand markCommand = new MarkCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage = String.format(MarkCommand.MESSAGE_MARK_PERSON_SUCCESS,
+                Messages.format(markedPerson));
+
+        ModelManager expectedModel = new ModelManager(model.getStaffBook(), new UserPrefs());
+        expectedModel.setPerson(model.getFilteredPersonList().get(0), markedPerson);
+
+        assertCommandSuccess(markCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        MarkCommand markCommand = new MarkCommand(outOfBoundIndex);
+
+        assertCommandFailure(markCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+
+        Person personToMark = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person markedPerson = new PersonBuilder(personToMark).withFavourite(true).build();
+        MarkCommand markCommand = new MarkCommand(INDEX_FIRST_PERSON);
+
+        String expectedMessage = String.format(MarkCommand.MESSAGE_MARK_PERSON_SUCCESS,
+                Messages.format(markedPerson));
+
+        Model expectedModel = new ModelManager(new StaffBook(model.getStaffBook()), new UserPrefs());
+        expectedModel.setPerson(model.getFilteredPersonList().get(0), markedPerson);
+
+        assertCommandSuccess(markCommand, model, expectedMessage, expectedModel);
+    }
```

Comment:
apologies if this should be obvious to me, but i was wondering if there is a difference between the `UnfilteredList_success()` and `FilteredList_success()` tests?

Reply:
For `FilteredList_success()` test, `showPersonAtIndex(model, INDEX_FIRST_PERSON)` is called to only display the person at the given index in this case `INDEX_FIRST_PERSON`, in the `model`'s staff book. Whereas for `UnfilteredList_success` test, the list is untouched at the beginning. Hope this answers your question. 😄